src/plugins/customer-order/generate-order/generate-customer-order.service.ts

```diff
@@ -0,0 +1,51 @@
+import {
+    Injectable,
+} from '@nestjs/common';
+import { 
+    TransactionalConnection,
+    RequestContext, 
+    ProductVariant 
+} from '@vendure/core';
+
+
+
+@Injectable()
+export class GenerateCustomerOrderService {
+    private relations = ['featuredAsset', 'taxCategory', 'channels'];
+    constructor(
+        private connection: TransactionalConnection
+    ) {}
+
+    /**
+     * Search for products from an array of codes
+     * @param ctx Session context
+     * @param inCodes  Array<String> codes to find in DataBase
+     */
+    async findProductVariantsBySKU(ctx: RequestContext, inCodes: Array<String>): Promise<Array<ProductVariant>> {
+        const qb = this.connection.getRepository<ProductVariant>(ctx,ProductVariant)
+        .createQueryBuilder('ProductVariant');
+        if (inCodes.length > 0) {
+            qb.andWhere(`ProductVariant.sku IN ( :...skus )`,{skus: inCodes});
+        } else {
+            qb.andWhere("ProductVariant.sku IN ('-1')");
+        }
+        qb.andWhere('ProductVariant.deletedAt IS NULL');
+        
+        return qb.getMany();
+    }
+
+
+    /**
+     * Validate that at least one product has stock to assign
+     * @param productVariants Array<ProductVariant>
+     */
+    isStockAvailable(productVariants: Array<ProductVariant>): Boolean {
+        let isAvailable = false;
+        productVariants.map((productVariant) => {
```

Comment:
No necesitas un map, con un `productVariants.forEach` es suficiente,  o un `for`, y haciendole un break cuando un producto este dsiponible.

Reply:
Incluso puedes hacer un `some`
https://developer.mozilla.org/es/docs/Web/JavaScript/Reference/Global_Objects/Array/some
```suggestion
        productVariants.some(({stockOnHand, stockAllocated}) => (stockOnHand -stockAllocated) > 0);
```